Diagnostics/ExchangeLogCollector/RemoteScriptBlock/IO/Save-WindowsEventLogs.ps1

```diff
@@ -14,15 +15,52 @@ Function Save-WindowsEventLogs {
             $_.VersionInfo.FileName
         }
 
+    $baseRegistryLocation = "SYSTEM\CurrentControlSet\Services\EventLog\"
+    $logs = @()
+    $baseParams = @{
+        MachineName = $env:COMPUTERNAME
+        GetValue    = "File"
+    }
+
     if ($PassedInfo.AppSysLogs) {
         Write-Verbose("Adding Application and System Logs")
-        $logs = @()
-        $logs += "$rootLogPath\Application.evtx"
-        $logs += "$rootLogPath\System.evtx"
-        $logs += "$rootLogPath\MSExchange Management.evtx"
+
+        $appParams = $baseParams + @{
+            SubKey = "$baseRegistryLocation`Application"
+        }
+        $sysParams = $baseParams + @{
+            SubKey = "$baseRegistryLocation`System"
+        }
+        $manParams = $baseParams + @{
+            SubKey = "$baseRegistryLocation`MSExchange Management"
+        }
+        $applicationLogs = Get-RemoteRegistryValue @appParams
+        $systemLogs = Get-RemoteRegistryValue @sysParams
+        $managementLogs = Get-RemoteRegistryValue @manParams
+
+        if ($null -eq $applicationLogs) { $applicationLogs = "$rootLogPath\Application.evtx" }
+
+        if ($null -eq $systemLogs) { $systemLogs = "$rootLogPath\System.evtx" }
+
+        if ($null -eq $managementLogs) { $managementLogs = "$rootLogPath\MSExchange Management.evtx" }
+
+        $logs += $applicationLogs
+        $logs += $systemLogs
+        $logs += $managementLogs
```

Comment:
The repetition here is bugging me. What about:

```powershell
foreach ($logName in @("Application", "System", "MSExchange Management")) {
  $regValue = Get-RemoteRegistryValue "$baseRegistryLocation$logName"
  $logs += ($null -ne $regValue ? $regValue : "$rootLogPath\$logName.evtx")
}
```

Reply:
would have to change it to an `if` statement not a `?`. Only available in PowerShell Core.

Reply:
how does this look now?

Reply:
Better.